From e1322be4d6416259a0e3e1e4370e8ff722cda146 Mon Sep 17 00:00:00 2001
From: Trek <trek00@inbox.ru>
Date: Mon, 1 Oct 2018 06:38:35 +0200
Subject: [PATCH] Wait for udev to be ready before trigger under sysvinit

The start-stop-daemon command with the --background argument returns
immediately, too soon to trigger events on some systems. Update the
SysV init script to wait until udev is ready, checking the control file.

Closes: #908796
---
 debian/udev.init | 13 +++++++++----
 1 file changed, 9 insertions(+), 4 deletions(-)

diff --git a/debian/udev.init b/debian/udev.init
index 9c394bb..374df4e 100644
--- a/debian/udev.init
+++ b/debian/udev.init
@@ -170,11 +170,16 @@ case "$1" in
         # prevents udevd to be killed by sendsigs (see #791944)
         mkdir -p $OMITDIR
         ln -sf $PIDFILE $OMITDIR/$NAME
-        log_end_msg $?
+
+        # wait for udev to be ready (see #908796)
+        timeout=150
+        until [ -S $CTRLFILE -o $timeout -le 0 ]; do
+            timeout=$((timeout - 1))
+            sleep 0.1
+        done
+        [ -S $CTRLFILE ] && log_success_msg || log_failure_msg
     else
-        log_warning_msg $?
-        log_warning_msg "Waiting 15 seconds and trying to continue anyway"
-        sleep 15
+        log_failure_msg
     fi
 
     log_action_begin_msg "Synthesizing the initial hotplug events"
-- 
2.1.4

